test/core/bad_client/tests/bad_streaming_id.cc

```diff
@@ -0,0 +1,132 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include <grpc/support/string_util.h>
+#include "src/core/lib/surface/server.h"
+#include "test/core/bad_client/bad_client.h"
+
+#define HEADER_FRAME_ID_1                                                  \
```

Comment:
Would it be better to place this in the test function to make it clearer? like

```
TEST(BadStreamingId, RegularHeader) {
  grpc_bad_client_arg args[2];
  args[0] = connection_preface_arg;
  args[1].client_validator = nullptr;
  SET_CLIENT_PAYLOAD(
     args[1],  
     "~~~~")
  grpc_run_bad_client_test(verifier, args, 2, GRPC_BAD_CLIENT_DISCONNECT);
}
```
Also you can use C preprocessor to concat given string literals. [link](https://stackoverflow.com/questions/5106280/string-concatenation-using-preprocessor)

Reply:
To avoid code duplication, i made it a macro. I intended to use the same frames at multiple places.

Which string literals are you suggesting concatenation for?

Reply:
Ah you already used it for concatenation.

```
 "HEADER"                                                                 \
 "\x10\x07:scheme\x04http"                                                \
 "\x10\x07:method\x04POST"                                                \
```

Literal \ had to be used here because it's part of macro but it might not be necessary once it's used a parameter for a macro.

Reply:
I'm still not sure what change you are suggesting :)